### PR TITLE
TextureResizeEncoder

### PR DESCRIPTION
**Pull request by eugenebokhan**

### CODE REVIEW

File: Alloy/Shaders/Shaders.metal [hunk 1]
```diff
@@ -52,6 +52,44 @@ generateKernels(textureCopy)
...
+                   const ushort2 position,
+                   const ushort2 totalThreads) {
+    const ushort2 textureSize = ushort2(destinationTexture.get_width(),
+                                        destinationTexture.get_height());
+    checkPosition(position, textureSize, deviceSupportsNonuniformThreadgroups);
+
+    const float2 normalizedCoord = float2((float(position.x) + 0.5) / textureSize.x,
+                                          (float(position.y) + 0.5) / textureSize.y);
+
+    const auto sampledlValue = sourceTexture.sample(sampler, normalizedCoord);
```
Comment (s1ddok) [inline Alloy/Shaders/Shaders.metal]: typo

File: Alloy/Encoders/TextureResizeEncoder.swift [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                        using: encoder)
+        }
+    }
+
+    public func encode(sourceTexture: MTLTexture,
+                       destinationTexture: MTLTexture,
+                       using encoder: MTLComputeCommandEncoder) {
+        encoder.set(textures: [sourceTexture, destinationTexture])
+
+        let sampler = encoder.device.makeSamplerState(descriptor: self.samplerDescriptor)
```
Comment (s1ddok) [inline Alloy/Encoders/TextureResizeEncoder.swift]: why not make it in init?

**Comment by s1ddok** [comment]
why do we need it in the first place when MPS covers the need?

**Comment by eugenebokhan** [comment]
We might need nearest sampling one day

**Pull request merged**